test/questions_test.py

```diff
@@ -19,31 +19,36 @@ def test_no_facets_then_return_no_questions(self):
 
     def test_one_discriminating_facet_then_return_one_element_question(self):
         question_generator = QuestionGenerator()
-        facet_a = Facet('NameA', 'ValueA')
+        facet_a1 = Facet('NameA', 'ValueA1')
+        facet_a2 = Facet('NameA', 'ValueA2')
         facet_b = Facet('NameB', 'ValueB')
+        facets_by_document = {'1': [facet_a1], '2': [facet_a2], '3': [facet_a2], '4': [facet_b]}
 
-        facets_by_document = {'1': [facet_a, facet_b], '2': [facet_a]}
         questions = question_generator.generate_questions(facets_by_document)
         self.assertEqual(len(questions), 1)
-        self.assertEqual(questions[0].facet_name, "NameB")
-        self.assertEqual(sorted(questions[0].facet_values), ['ValueB'])
+        self.assertEqual(questions[0].facet_name, "NameA")
+        self.assertEqual(sorted(questions[0].facet_values), ['ValueA1', 'ValueA2'])
 
     def test_similar_values_different_facets_then_return_one_question(self):
         question_generator = QuestionGenerator()
-        facet_a = Facet('NameA', 'ValueA')
+        facet_a1 = Facet('NameA', 'ValueA')
         facet_a2 = Facet('NameA', 'ValueA')
-        facet_b = Facet('NameB', 'ValueB')
-        facets_by_document = {'1': [facet_a2, facet_b], '2': [facet_a]}
```

Comment:
For this test, we want to see if filtering is done using the content of the object instead of the reference. A better option would be:
`facet_a = Facet('NameA', 'ValueA')`
`facet_a2 = Facet('NameA', 'ValueA')`
`facet_b = Facet('NameB', 'ValueB')`
`facets_by_document = {'1': [facet_a2, facet_b], '2': [facet_a], '3': [facet_a]}`